demomultistoreform/src/Form/ContentBlockType.php

```diff
@@ -23,34 +23,21 @@
 
 use PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType;
 use PrestaShopBundle\Form\Admin\Type\SwitchType;
-use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
 use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\TypedRegex;
-use Symfony\Component\Translation\TranslatorInterface;
 use Symfony\Component\Validator\Constraints\Length;
 use Symfony\Component\Validator\Constraints\NotBlank;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
 
-class ContentBlockType extends TranslatorAwareType
+class ContentBlockType extends AbstractType
 {
-    /**
-     * @var bool
-     */
-    private $isMultistoreUsed;
-
-    /**
-     * @param TranslatorInterface $translator
-     * @param array $locales
-     * @param bool $isMultistoreUsed
-     */
+    use TranslatorAwareTrait;
+    
     public function __construct(
-        TranslatorInterface $translator,
-        array $locales,
-        bool $isMultistoreUsed
+        private readonly bool $isMultistoreUsed
```

Comment:
Not required but similmarly here you could inject `ShopContext` instead of some logic in the YAML definition to get a boolean `'@=service("prestashop.adapter.multistore_feature").isUsed()'`